components/logins/android/src/main/java/mozilla/appservices/logins/DatabaseLoginsStorage.kt

```diff
@@ -366,6 +372,19 @@ internal fun Pointer.getAndConsumeRustString(): String {
     }
 }
 
+/**
+ * Helper to read a protocal buffer.
+ *
+ * Important: Do not use this buffer after this!
+ */
+internal fun RustBuffer.ByValue.getAndConsumeBuffer(): CodedInputStream? {
```

Comment:
Can we inline this in the functions above, so we can wrap the `parseFrom` calls in the `try`?